proposals/p2360.md

```diff
@@ -0,0 +1,291 @@
+# Types are values of type `Type`
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/2360)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Problem](#problem)
+    -   [Tuples behave inconsistently](#tuples-behave-inconsistently)
+    -   [Qualified access to constraint members is inconsistent](#qualified-access-to-constraint-members-is-inconsistent)
+-   [Background](#background)
+-   [Proposal](#proposal)
+-   [Details](#details)
+    -   [Terminology](#terminology)
+    -   [Compound member access](#compound-member-access)
+    -   [Tuple and struct literals](#tuple-and-struct-literals)
+    -   [Name lookup into values with constrained types](#name-lookup-into-values-with-constrained-types)
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+    -   [Alternative terminology](#alternative-terminology)
+
+<!-- tocstop -->
+
+## Abstract
+
+Define a "type" to be a value of type `Type`. Contexts expecting a type perform
+an implicit conversion to `Type`. Values like `()` and `(i32, i32)` and `{}` are
+no longer types, but instead implicitly convert to `Type`. Values of interface
+or constraint type (now called "kinds") are similarly not formally types but
+implicitly convert to `Type`.
```

Comment:
Maybe slightly expand this to make it clear that this has a relatively small *change* to the design, and is more about terminology and more crisp concepts with some more minor changes to address narrow problems and make things more consistent?

```suggestion
## Abstract

Define a "type" to be a value of type `Type`. Contexts expecting a type perform
an implicit conversion to `Type`. Values like `()` and `(i32, i32)` and `{}` are
no longer types, but instead implicitly convert to `Type`. Values of interface
or constraint type (now called "kinds") are similarly not formally types but
implicitly convert to `Type`.

The practical impact to the semantics are relatively small. This is primarily
establishing more principled and clear terminology and concepts in the design
with a few more narrow fixes and improvements to the design's consistency.
```

Reply:
Rephrased a bit to avoid repeated `and`.